Update README.md with PR note (#1853)
w3c/wot-thing-description: README.md

> __Note (August 2022):__

Author: I am unsure "why" the note still says August 2022 ?

Reviewer: ```suggestion
__Note (July 2023):__
```

> __Please notice that Thing Description 1.1 is in the review phase for the Proposed Recommendation (PR) transition. 

Reviewer: ```suggestion
__Please notice that Thing Description 1.1 is in the review phase for the Recommendation transition. 
```

We are done with PR transition so now we are in the rec transition phase.

> In this and subsequent phase, only editorial changes (e.g., adding examples, additional explanations) and bug fixes can be applied to the [editorial draft](http://w3c.github.io/wot-thing-description/). Editorial changes should be labeled with 'Editorial' in the issue and corresponding PRs. New feature requests can not be considered and will be postponed to the next version of the Thing Description.__  

Reviewer: ```suggestion
In this phase, only editorial changes (e.g., adding examples, additional explanations) and bug fixes can be applied to the [editorial draft](http://w3c.github.io/wot-thing-description/) if the whole Working Group agrees on it. Editorial changes should be labeled with 'Editorial' in the issue and corresponding PRs. New feature requests can not be considered and will be postponed to the next version of the Thing Description.__  
```
